### PR TITLE
Head Start and Early Head Start eligibility 

### LINKED ISSUE #150: Head Start and Early Head Start eligibility
See:
* https://www.atlantafed.org/-/media/documents/economic-mobility-and-resilience/advancing-careers-for-low-income-families/policy-rules-database/policy-rules-database-technical-manual.pdf#page=58
* https://eclkc.ohs.acf.hhs.gov/policy/head-start-act/sec-645-participation-head-start-programs
* https://eclkc.ohs.acf.hhs.gov/data-ongoing-monitoring/article/program-information-report-pir
* [$10.7 billion budget](https://eclkc.ohs.acf.hhs.gov/policy/pi/acf-pi-hs-21-01)
* https://www.law.cornell.edu/uscode/text/42/9840#a_1

### LINKED ISSUE #4735: Expand free school categorical eligibility to Head Start children
Will not be included in #4719 

References: 
https://www.fns.usda.gov/cn/extending-categorical-eligibility-additional-children-household
https://www.law.cornell.edu/uscode/text/42/1758#b_5

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #150 
Fixes #4735

### CODE REVIEW

**Comment by vrathi101** [comment]
I used childcare_hours_per_week variable and multiplied by 52 instead of trying to find an "average per year statistic"

File: policyengine_us/parameters/gov/hhs/head_start/age_range.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+description: The Administration for Children and Families limits Head Start participants to children within this age range.
+
+brackets:
+  - threshold:
+      2021-09-01: 3
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/age_range.yaml:5]: this should start at 0

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/hhs/head_start/categorical_eligibility.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    - tanf
+    - ssi
+    - snap
+    - is_homeless
+    - was_in_foster_care
+
+metadata:
+  unit: list
+  period: year
+  label: Qualifying programs for Head Start or Early Head Start participation.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/categorical_eligibility.yaml:13]:
```suggestion
  label: Head Start qualifying programs
```

File: policyengine_us/parameters/gov/hhs/head_start/early_head_start/age_limit.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+description: The Administration for Children and Families limits the Early Head Start participants to children below this age limit.
+
+values:
+  2021-09-01: 3
+
+metadata:
+  unit: year
+  period: year
+  label: Early Head Start children age maximum
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/early_head_start/age_limit.yaml:9]:
```suggestion
  label: Early Head Start children age limit
```

File: policyengine_us/variables/gov/hhs/head_start/is_early_head_start_eligible.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from policyengine_us.model_api import *
+
+
+class is_early_head_start_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible for the Early Head Start program"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/hhs/head_start/is_early_head_start_eligible.py:7]:
```suggestion
    label = "Eligible person for the Early Head Start program"
```

File: policyengine_us/variables/gov/hhs/head_start/is_head_start_eligible.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+from policyengine_us.model_api import *
+
+
+class is_head_start_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible for the Head Start program"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/hhs/head_start/is_head_start_eligible.py:7]:
```suggestion
    label = "Eligible person for the Head Start program"
```

File: policyengine_us/parameters/gov/hhs/head_start/early_head_start/minimum_hours.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+description: The Administration for Children and Families mandates that the Early Head Start program provides a minimum of this many annual class operation hours for its enrolled children.
+values:
+  2021-09-01: 1_380
+
+metadata:
+  unit: hour
+  period: year
+  label: Early Head Start minimum annual hours of operation
+  reference:
+    - title: 1302.21 Center-based option.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/early_head_start/minimum_hours.yaml:10]:
```suggestion
    - title: U.S. Department of Health & Human Services Head Start Policy and Regulations 1302.21 Center-based option
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/hhs/head_start/early_head_start/minimum_hours.yaml:10]: all titles

File: policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml:1]: for the value of the program we would need to see how we can quantify these hours - I think this is less of an eligibility requirement for the people looking to enroll but an requirement for the program to meet to be qualified as a head start program

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml
Comment (vrathi101) [reply policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml:1]: > for the value of the program we would need to see how we can quantify these hours - I think this is less of an eligibility requirement for the people looking to enroll but an requirement for the program to meet to be qualified as a head start program

yeah these hours weren't included in eligibility tho, they were used to estimate what fraction of a family's childcare hours would be taken care of by head start program hours, am I missing something? that's why its used in head_start.py and early_head_start.py

File: policyengine_us/parameters/gov/hhs/head_start/early_head_start/minimum_hours.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/early_head_start/minimum_hours.yaml:1]: i think we can remove this for now

File: policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/hhs/head_start/minimum_hours.yaml:1]: we can also remove this

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml [hunk 7]
```diff
@@ -4,14 +4,13 @@ values:
...
     # Also not implemented:
-    # - Federal Public Housing Assistance
     # - Veterans and Survivors Pension Benefit
     # - Tribal programs:
     #   - Bureau of Indian Affairs general assistance
-    #   - Tribally administered Temporary Assistance for Needy Families
-    #   - Head Start (only those households meeting its income qualifying standard)
-    #   - Food Distribution Program on Indian Reservations
 metadata:
   period: month
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml:15]: while we are in here:
add label and unit

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by MaxGhenis** [comment]
Can we fix CI? The recurrent windows failure is new.

**Comment by vrathi101** [comment]
> Can we fix CI? The recurrent windows failure is new.

working now

File: policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml [hunk 8]
```diff
@@ -4,16 +4,17 @@ values:
     - medicaid
     - snap
     - ssi
+    - is_head_start_income_eligible
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml:7]: these three actually only apply to people on tribal lands. needs separate logic.

we also need to base head start categorical eligibility on participation (`head_start`) not only the income eligibility

File: policyengine_us/parameters/gov/states/ca/cpuc/care/eligibility/categorical.yaml [hunk 9]
```diff
@@ -6,11 +6,11 @@ values:
     - snap
     - ssi
     - tanf
+    - is_head_start_income_eligible
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ca/cpuc/care/eligibility/categorical.yaml:9]: similarly, needs to be based on participation

File: policyengine_us/variables/gov/hhs/head_start/is_head_start_program_eligible.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class is_head_start_program_eligible(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/hhs/head_start/is_head_start_program_eligible.py:4]:
```suggestion
class is_head_start_categorically_eligible(Variable):
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml (hunk 8 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml:7]: > these three actually only apply to people on tribal lands. needs separate logic.
> 
> we also need to base head start categorical eligibility on participation (`head_start`) not only the income eligibility

regarding your second point, the legal code states "only those [households] meeting its income qualifying standard", wouldn't that mean specifically head start income eligible

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml:7]: yes, but it also says they must participate. someone income-eligible but who doesn't participate wouldn't qualify

Comment (vrathi101) [reply policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml]: this PR only covers head start eligibility as we weren't able to find any explicit quantitative benefits of the program, for the categorical eligibility files I changed it to is_head_start_eligible but should I leave them out until we develop a head_start variable?

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/fcc/lifeline/categorical_eligibility.yaml]: Please either leave it out or point it to a head_start variable that doesn't yet have a formula. Pointing it to eligibility indicates a different policy.

File: policyengine_us/variables/gov/hhs/head_start/head_start.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
+from policyengine_us.model_api import *
+
+
+class head_start(Variable):
+    value_type = float
+    entity = Person
+    label = "Amount of Head Start benefit"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/hhs/head_start/head_start.py:10]: remove method. we let users enter values for variables without formulas

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**